specs/_features/eip7594/polynomial-commitments-sampling.md

```diff
@@ -228,46 +223,29 @@ def coset_fft_field(vals: Sequence[BLSFieldElement],
 #### `compute_verify_cell_kzg_proof_batch_challenge`
 
 ```python
-def compute_verify_cell_kzg_proof_batch_challenge(row_commitments: Sequence[KZGCommitment],
-                                                  row_indices: Sequence[RowIndex],
-                                                  column_indices: Sequence[ColumnIndex],
+def compute_verify_cell_kzg_proof_batch_challenge(commitments: Sequence[KZGCommitment],
+                                                  commitment_indices: Sequence[CommitmentIndex],
+                                                  cell_indices: Sequence[CellIndex],
                                                   cosets_evals: Sequence[CosetEvals],
                                                   proofs: Sequence[KZGProof]) -> BLSFieldElement:
     """
-    Compute a random challenge r used in the universal verification equation.
-    This is used in verify_cell_kzg_proof_batch_impl.
-
-    To compute the challenge, `RANDOM_CHALLENGE_KZG_CELL_BATCH_DOMAIN` is used as a hash prefix.
+    Compute a random challenge ``r`` used in the universal verification equation. To compute the
+    challenge, ``RANDOM_CHALLENGE_KZG_CELL_BATCH_DOMAIN`` and all data that can influence the
+    verification is hashed together to generate a deterministicly "random" BLS field element.
     """
-    # input the domain separator
     hashinput = RANDOM_CHALLENGE_KZG_CELL_BATCH_DOMAIN
-
-    # input the degree bound of the polynomial
     hashinput += int.to_bytes(FIELD_ELEMENTS_PER_BLOB, 8, KZG_ENDIANNESS)
-
-    # input the field elements per cell
     hashinput += int.to_bytes(FIELD_ELEMENTS_PER_CELL, 8, KZG_ENDIANNESS)
-
-    # input the number of commitments
-    num_commitments = len(row_commitments)
-    hashinput += int.to_bytes(num_commitments, 8, KZG_ENDIANNESS)
-
-    # input the number of cells
-    num_cells = len(row_indices)
-    hashinput += int.to_bytes(num_cells, 8, KZG_ENDIANNESS)
-
-    # input all commitments
-    for commitment in row_commitments:
+    hashinput += int.to_bytes(len(commitments), 8, KZG_ENDIANNESS)
+    hashinput += int.to_bytes(len(cosets_evals), 8, KZG_ENDIANNESS)
```

Comment:
I feel like it's cleaner to refer to `cell_indices` here, since we want to insert the number of cells. I had to make sure that `len(cosets_evals) == len(cell_indices)`.